appcelerator/titanium_mobile
# [TIDOC-2143] Add addon generator and some refactors and fixes

Testing instructions:

Install dependencies.  From the apidoc directory, run:
`npm install .`

Locally pull in changes from this PR.
Generate add on docs for BlackBerry:
`node docgen -f addon -p blackberry`

Locally pull in changes from #6672 to remove the BlackBerry docs from the core documentation.

Generate docs with addon docs:
`node docgen -a ~/titanium_mobile/dist/addon`

Open ~/titanium_mobile/dist/apidoc/index.html and spot check the docs. BlackBerry should appear in the same places as before.

You can also locally test changes from https://github.com/appcelerator/titanium_mobile_blackberry/pull/275
